src/seedu/addressbook/commands/FindCommand.java

```diff
@@ -50,11 +50,29 @@ public CommandResult execute() {
         final List<ReadOnlyPerson> matchedPersons = new ArrayList<>();
         for (ReadOnlyPerson person : addressBook.getAllPersons()) {
             final Set<String> wordsInName = new HashSet<>(person.getName().getWordsInName());
-            if (!Collections.disjoint(wordsInName, keywords)) {
+            if (hasSameCharacters(wordsInName, keywords)) {
```

Comment:
From the method name alone, I can assume that there is no conversion to lowercase. So, may be it will be better if the conversion is done outside `hasSameCharacters` method.